pkg/avancement/service_manager.go

```diff
@@ -134,23 +146,29 @@ func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName stri
 func (s *ServiceManager) checkoutSourceRepo(repoURL, branch string) (git.Repo, error) {
 	repo, err := s.cloneRepo(repoURL, branch)
 	if err != nil {
-		return nil, fmt.Errorf("failed to clone source repo %s: %w", repoURL, err)
+		if git.IsGitError(err) {
+			return nil, git.GitError("failed to clone source repository", repoURL)
+		}
+		return nil, err
 	}
 	err = repo.Checkout(branch)
 	if err != nil {
-		return nil, fmt.Errorf("failed to checkout branch %s in repo %s: %w", branch, repoURL, err)
+		return nil, git.GitError(fmt.Sprintf("failed to checkout branch %s", branch), repoURL)
```

Comment:
This is losing the actual error message.

And due to the lack of wrapping, it wouldn't even be possible to work out which bit was failing.

Reply:
I've seen way too many cases where you do something like...

```shell
$ ./run-this-command --with-this --and-that /tmp/myfile
open config.txt: no such file or directory
```
And you're left wondering where and why it was trying to open the file.

Reply:
Yeah, you're right, so surfacing this real err is definitely a safe one to do (see my other comment about the repo URL where I think it might not be but i'm gonna confirm that)